platform/rpc/nats/bus.go

```diff
@@ -342,6 +349,11 @@ func (n *NATS) Subscribe(instID flux.InstanceID, remote platform.Platform, done
 				// dispatch in a goroutine and deliver any errors back to us so that we can
 				// clean up on any hard failures.
 				go processRequest(request)
+			case <-forceReconnect.C:
+				sub.Unsubscribe()
+				close(requests)
+				done <- nil
+				return
```

Comment:
Would be nice to ensure there were not outstanding requests when this happens... but that is trickier...